### ISSUE TITLE
Jacobi symbol incomplete

### ISSUE THREAD

**Issue by czurnieden** [libtom/libtommath#44]
The the algorithm to compute the Jacobi is incomplete: the domain of the numerator of the Jacobi symbol is the whole Z. Negative numerators can be computed with the help of quadratic reciprocity, that is: (n/m) = (-1/n)(n/m). The value of (-1/n) comes from the Legendre symbol (1) and is (-1)^((n-1)/2). The lowest two bits of n are enough to compute this factor but working with the whole lowest limb might be more legible.

A rough sketch using the whole lowest limb:

```
 /* at the end */
 tmp = p->dp[0] - 1;
 tmp >>= 1;
/* change sign if "tmp" is odd */
 tmp = (tmp & 1) ? 1 : -1;
 /* if "a" is unity "tmp" is already the correct result */
 /* use "tmp" to define the sign of the result before returning */
```

(1) from the congruence a^((p-1)/2) = (a/p) mod p
see for example: Henri Cohen "A Course in Computational Algebraic Number Theory" of which you might find a copy on the internet.

**Comment by sjaeckel** [comment]
can this be closed when #113 is merged? :)

**Comment by czurnieden** [comment]
Would be obsoleted by #113, yes.
